### PR TITLE
Improve error logging of elastic search telemetry module

### PR DESCRIPTION
**Pull request by tsachiherman**
## Summary

This PR addresses several key locations which lacked error message printouts.

## Test Plan

These changes were tested manually. Automating a test for these seems to be quite an intensive task, as it requires to emulate an elastic search server..

### CODE REVIEW

File: logging/telemetryhook.go [hunk 1]
```diff
@@ -87,7 +86,10 @@ func createAsyncHookLevels(wrappedHook logrus.Hook, channelDepth uint, maxQueueD
 					}
 					hook.Unlock()
 					if entry != nil {
-						hook.wrappedHook.Fire(entry)
+						err := hook.wrappedHook.Fire(entry)
```
Comment (algonautshant) [inline logging/telemetryhook.go:89]: Do we also need to capture the error returned from Fire here (in func logTelemetry)?
logging/telemetry.go 228:	t.hook.Fire(entry)

**Review by algonautshant — APPROVED** [review]
Looks good. 
Just one comment.

File: logging/telemetryhook.go [hunk 2]
```diff
@@ -199,6 +202,24 @@ func (hook *dummyHook) waitForEventAndReady() bool {
 	return true
 }
 
+type elasticClientLogger struct {
+	elastic.Logger
```
Comment (brianolson) [inline logging/telemetryhook.go:206]: to be a elastic logger just needs Printf, which always points to Base(), so I think we don't need this part of the struct.

File: logging/telemetryhook.go (hunk 1 above)
Comment (tsachiherman) [reply logging/telemetryhook.go:89]: The Fire function on the `logging/telemetry.go 228` goes to `asyncTelemetryHook.Fire` which aways returns nil, so adding a logging there is correct, but won't really be helpful.

**Review by brianolson — APPROVED** [review]

**Pull request merged**